featurebyte/models/feature_table_cache.py

```diff
@@ -0,0 +1,55 @@
+"""
+This module contains Feature Table Cache related models
+"""
+from typing import List
+
+import pymongo
+
+from featurebyte.models.base import (
+    FeatureByteCatalogBaseDocumentModel,
+    PydanticObjectId,
+    UniqueConstraintResolutionSignature,
+    UniqueValuesConstraint,
+)
+
+
+class FeatureTableCacheModel(FeatureByteCatalogBaseDocumentModel):
+    """
+    Model for feature table cache
+
+    id: PydanticObjectId
+        FeatureTableCache id of the object
+    observation_table_id: PydanticObjectId
+        Observation Table Id
+    name: str
+        Name of the Feature Table Cache. Also used as name of the table in DWH
+    features: List[str]
+        List of feature names which are stored in this feature table cache
+    """
+
+    observation_table_id: PydanticObjectId
+    name: str
+    features: List[str]
```

Comment:
Could it be better if we use the schema like
```
features: List[FeatureInfo]
```

where `FeatureInfo` is something like
```
class FeatureInfo:
    name: str
    feature_id: PydanticObjectId
```

Reply:
We don't really what it to be based feature id, because the goal is to be able to materialise ideated features before they were actually saved as features

Reply:
The main purpose of storing feature ID is to trace the source of the cache. Would be good if we could trace whether it is from saved feature or ideated feature.

Reply:
This makes sense. What about the case when multiple features with different names and ids have same definition hash? We were hoping to cache such feature once per observation table, whereas adding feature id might be misleading. 
Do we need to worry about this or is it better to just ignore this case for now?

Reply:
It is just for the troubleshooting purpose. We only need to track the feature info when the cache is created, right? Just want to know where the value comes from as feature definition hash has too little information to trace. We could definitely add this later.

Reply:
Yes sound good, added the feature id